runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -126,66 +124,64 @@ public Layout prepare(long epoch, long rank)
                     return cf;
                 })
                 .toArray(CompletableFuture[]::new);
-        LayoutPrepareResponse[] acceptList;
+
+        long promises = 0L;
         long timeouts = 0L;
+        long outranks = 0L;
         long wrongEpochRejected = 0L;
-        while (true) {
```

Comment:
This this while `loop` has been removed, are you assuming that upper layers will retry the phases again ?
Could this impact latency, when two fault detectors detect the same error and try to compete on updating the layout ?

Reply:
Yes, I am assuming that, and I think this while loop assumes too.. While loop doesn't retry either.